### PR TITLE
new: Image mark

### PR DESCRIPTION
**Pull request by maartenbreddels**
Usage
```python
import bqplot
import ipywidgets as widgets

with open("sample.png", 'rb') as f:
    data = f.read()
image = widgets.Image(value=data)

scale_x = bqplot.LinearScale(min=-1, max=2)
scale_y = bqplot.LinearScale(min=-0.5, max=2)
image_mark = bqplot.Image(image=image, scales={'x': scale_x, 'y': scale_y})
fig = bqplot.Figure(padding_x=0, padding_y=0)
fig.marks = [image_mark]
fig.axes = [bqplot.Axis(scale=scale_x), bqplot.Axis(scale=scale_y, orientation='vertical')]
fig
```
It uses the widget.Image widget. Maybe it doesn't do all the things, like colorscales and all, but I think this makes a lot possible already.

### CODE REVIEW

**Comment by dmadeka** [comment]
@maartenbreddels Awesome

**Comment by ssunkara1** [comment]
@maartenbreddels Somehow I cannot get the `Image` mark to work with your example above. I can get `ipywidgets.Image` to display fine in the output area but not on a `Figure`.

**Comment by maartenbreddels** [comment]
Are you using ipywidgets7?

**Comment by ssunkara1** [comment]
Yeah I have `ipywidgets 7.0.0b1`, `widgetsnbextension 3.0.0b1`

**Comment by maartenbreddels** [comment]
Can you show me some error msg'es from the javascript console?

File: js/src/Image.js [hunk 1]
```diff
@@ -0,0 +1,97 @@
+var d3 = require("d3");
```
Comment (jasongrout) [inline js/src/Image.js]: Missing copyright

**Comment by jasongrout** [comment]
I'm curious - why use the ipywidgets image here instead of just having the image source directly? Does setting image layout attributes mess this up?

**Comment by maartenbreddels** [comment]
Well, i'm not sure actually what would be the best strategy, since it would be useful to have everything that ipywidgets' Image can have, especially after https://github.com/jupyter-widgets/ipywidgets/pull/1676 gets in. So either we copy all of that, or none of that and reuse Image. Other ideas are welcome.

**Comment by dmadeka** [comment]
@ssunkara1 Maybe we can get this in by PyData NYC?

File: js/src/ImageModel.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+            scales_metadata: {
+                'x': {'orientation': 'horizontal', 'dimension': 'x'},
+                'y': {'orientation': 'vertical', 'dimension': 'y'},
+            },
+        });
+    },
+
+    initialize: function() {
+        ImageModel.__super__.initialize.apply(this, arguments);
+        this.on("change:x0 change:y0 change:x1 change:y1", this.update_data, this);
```
Comment (rmenegaux) [inline js/src/ImageModel.js]: Any reason for not using `on_some_change`here?

**Comment by rmenegaux** [comment]
@maartenbreddels looks good to me. Any idea why travis is failing?

File: js/src/ImageModel.js (hunk 2 above)
Comment (maartenbreddels) [reply js/src/ImageModel.js]: No, didn't know it existed, let me change that

**Comment by maartenbreddels** [comment]
Just a network timeout for one of them, let me change that on_some_change and it will rebuild.

**Comment by maartenbreddels** [comment]
🎉

**Comment by outlace** [comment]
This example  is also not working for me, here's my Javascript console error output:
```python

manager-base.js:272 Could not instantiate widget
(anonymous) @ manager-base.js:272
step @ manager-base.js:42
(anonymous) @ manager-base.js:23
rejected @ manager-base.js:15
Promise rejected (async)
step @ manager-base.js:16
(anonymous) @ manager-base.js:17
__awaiter @ manager-base.js:13
ManagerBase._make_model @ manager-base.js:255
(anonymous) @ manager-base.js:244
step @ manager-base.js:42
(anonymous) @ manager-base.js:23
(anonymous) @ manager-base.js:17
__awaiter @ manager-base.js:13
ManagerBase.new_model @ manager-base.js:230
ManagerBase.handle_comm_open @ manager-base.js:149
(anonymous) @ comm.js:89
Promise resolved (async)
CommManager.comm_open @ comm.js:85
i @ jquery.min.js:4
Kernel._handle_iopub_message @ kernel.js:1223
Kernel._finish_ws_message @ kernel.js:1015
(anonymous) @ kernel.js:1006
Promise resolved (async)
Kernel._handle_ws_message @ kernel.js:1006
i @ jquery.min.js:4
utils.js:109 Error: Could not create a model.
    at promiseRejection (utils.js:109)
    at <anonymous>
promiseRejection @ utils.js:109
Promise rejected (async)
ManagerBase.handle_comm_open @ manager-base.js:154
(anonymous) @ comm.js:89
Promise resolved (async)
CommManager.comm_open @ comm.js:85
i @ jquery.min.js:4
Kernel._handle_iopub_message @ kernel.js:1223
Kernel._finish_ws_message @ kernel.js:1015
(anonymous) @ kernel.js:1006
Promise resolved (async)
Kernel._handle_ws_message @ kernel.js:1006
i @ jquery.min.js:4
kernel.js:1007 Couldn't process kernel message Class ImageModel not found in module bqplot
(anonymous) @ kernel.js:1007
Promise rejected (async)
Kernel._handle_ws_message @ kernel.js:1007
i @ jquery.min.js:4
utils.js:109 Error: Could not process update msg for model id: 325a891141a64996b48576a05312c9eb
    at promiseRejection (utils.js:109)
    at <anonymous>
promiseRejection @ utils.js:109
Promise rejected (async)
WidgetModel._handle_comm_msg @ widget.js:211
Comm._callback @ comm.js:194
Comm.handle_msg @ comm.js:202
(anonymous) @ comm.js:132
Promise resolved (async)
CommManager.comm_msg @ comm.js:131
i @ jquery.min.js:4
Kernel._handle_iopub_message @ kernel.js:1223
Kernel._finish_ws_message @ kernel.js:1015
(anonymous) @ kernel.js:1006
Promise resolved (async)
Kernel._handle_ws_message @ kernel.js:1006
i @ jquery.min.js:4
kernel.js:1007 Couldn't process kernel message WrappedError {stack: "Error: Exception handling comm message↵    at new …20f809e44ec2d50e40412:13277:33↵    at <anonymous>", message: "Exception handling comm message", error_stack: Array(2)}
(anonymous) @ kernel.js:1007
Promise rejected (async)
Kernel._handle_ws_message @ kernel.js:1007
i @ jquery.min.js:4
kernel.js:1007 Couldn't process kernel message WrappedError {stack: "Error: Exception handling comm message↵    at new …20f809e44ec2d50e40412:13277:33↵    at <anonymous>", message: "Exception handling comm message", error_stack: Array(2)}
(anonymous) @ kernel.js:1007
Promise rejected (async)
Kernel._handle_ws_message @ kernel.js:1007
i @ jquery.min.js:4
utils.js:109 Error: Could not create view
    at promiseRejection (utils.js:109)
    at <anonymous>
promiseRejection @ utils.js:109
Promise rejected (async)
ManagerBase.display_model @ manager-base.js:71
(anonymous) @ extension.js:114
Promise resolved (async)
render @ extension.js:113
append_mime @ extension.js:131
OutputArea.append_mime_type @ outputarea.js:671
OutputArea.append_display_data @ outputarea.js:637
OutputArea.append_output @ outputarea.js:330
OutputArea.handle_output @ outputarea.js:243
output @ codecell.js:365
Kernel._handle_output_message @ kernel.js:1196
i @ jquery.min.js:4
Kernel._handle_iopub_message @ kernel.js:1223
Kernel._finish_ws_message @ kernel.js:1015
(anonymous) @ kernel.js:1006
Promise resolved (async)
Kernel._handle_ws_message @ kernel.js:1006
i @ jquery.min.js:4
Ca%20Analysis%20v1.0-(Interactive).ipynb#:1 Uncaught (in promise) Class ImageModel not found in module bqplot
Promise rejected (async)
ManagerBase.create_view @ manager-base.js:104
ManagerBase.display_model @ manager-base.js:71
(anonymous) @ extension.js:114
Promise resolved (async)
render @ extension.js:113
append_mime @ extension.js:131
OutputArea.append_mime_type @ outputarea.js:671
OutputArea.append_display_data @ outputarea.js:637
OutputArea.append_output @ outputarea.js:330
OutputArea.handle_output @ outputarea.js:243
output @ codecell.js:365
Kernel._handle_output_message @ kernel.js:1196
i @ jquery.min.js:4
Kernel._handle_iopub_message @ kernel.js:1223
Kernel._finish_ws_message @ kernel.js:1015
(anonymous) @ kernel.js:1006
Promise resolved (async)
Kernel._handle_ws_message @ kernel.js:1006
i @ jquery.min.js:4
Ca%20Analysis%20v1.0-(Interactive).ipynb#:1 Uncaught (in promise) Class ImageModel not found in module bqplot

```

**Comment by SylvainCorlay** [comment]
@outlace did you install bqplot from source?

We have not released a version with that feature yet. This feature has just been merged recently.

**Comment by outlace** [comment]
Yes I built from source, followed the instructions for devs on the readme

**Comment by outlace** [comment]
I see this wants a bytes string, whats the easiest way to get this to work with an image in the form of  a numpy array?

**Pull request merged**